modules/dcache/src/main/java/org/dcache/pool/statistics/IoStatistics.java

```diff
@@ -65,14 +68,17 @@ public void updateRead(long readBytes, long readTime){
      *         The duration of the write request in nanoseconds; must be non-negative
      */
 
-    public void updateWrite(long writeBytes, long writeTime){
+    public void updateWrite(long writeBytes, long writeTime) throws ArrayIndexOutOfBoundsException{
+
         double writeSpeed = calculateSpeed(writeBytes, writeTime);
-        _writeSpeeds.add(writeSpeed);
+        _writeSpeeds[_writeSpeedsArrayIndex] = writeSpeed;
+        _writeSpeedsArrayIndex++;
```

Comment:
Sorry if we are making your life difficult :-)

But I think we're almost there  ...